camelot/view/controls/editors/one2manyeditor.py

```diff
@@ -139,14 +163,21 @@ def set_field_attributes(self, **kwargs):
         self.update_action_status()
 
     def update_action_status(self):
-        toolbar = self.findChild(QtWidgets.QToolBar)
-        if toolbar:
-            model_context = self.gui_context.create_model_context()
-            for qaction in toolbar.actions() + toolbar.findChildren(ActionToolbutton) + toolbar.findChildren(ActionPushButton):
-                if isinstance(qaction, (ActionAction, ActionToolbutton, ActionPushButton)):
-                    post(qaction.action.get_state,
-                         qaction.set_state,
-                         args=(model_context,))
+        table = self.gui_context.item_view
+        selection_model = table.selectionModel()
+        current_index = table.currentIndex()
+        table.model().change_selection(selection_model, current_index)
+
+    def current_row_changed(self, current=None, previous=None):
+        self.update_action_status()
+
+    @QtCore.qt_slot(tuple, State)
+    def action_state_changed(self, route, state):
+        action = AdminRoute.action_for(route)
```

Comment:
in de gui kan je in principe de action nt resolven uit de route, maar dat is ook nt nodig, gezien je action_widget ook op basis vd route kan zoeken

Reply:
done